docs/creating-a-static-check.md

```diff
@@ -0,0 +1,134 @@
+# Creating a Static Check
+
+Static checks are used to perform static analysis against a plugin's codebase.
+
+The Static checks, which analyze the code, either using PHP CodeSniffer sniffs or custom logic e.g. using regular expressions.
+
+```php
+use WordPress\Plugin_Check\Checker\Check_Result;
+
+class Custom_Check extends Static_Check {
+  public function run( Check_Result $result );
+    // Handle running the check and adding warnings or errors to the result.
+  }
```

Comment:
Apply similar feedback that i raise for `Runtime Check`

Reply:
Thanks @mukeshpanchal27 I've added this here